docker-compose.yml

```diff
@@ -1,49 +1,78 @@
-version: '3'
+version: "3.4"
+
 
 services:
-  web:
-    build: .
-    command: gunicorn --reload --bind=0.0.0.0:8000 eahub.config.wsgi
-    depends_on:
-      - db
-      - mail
-      - webdriver
-      - chrome
-      - firefox
-    ports:
-      - 8000:8000
-    links:
-      - 'db:postgres'
-    volumes:
-      - .:/app:rw
-      - ./data:/data:rw
-    env_file:
-      - .env
-  db:
-    image: postgres:9.6
-    ports:
-      - 5432:5432
-    environment:
-      POSTGRES_DB: 'db'
-      POSTGRES_HOST_AUTH_METHOD: trust
-    volumes:
-      - .:/app:rw
-  mail:
-    image: dockage/mailcatcher:0.6.5
-    ports:
-      - 1025:1025
-      - 1080:1080
-  webdriver:
-    image: selenium/hub:3.141.59-iron
-  chrome:
-    image: selenium/node-chrome:3.141.59-iron
-    depends_on:
-      - webdriver
-    environment:
-      HUB_HOST: webdriver
-  firefox:
-    image: selenium/node-firefox:3.141.59-iron
-    depends_on:
-      - webdriver
-    environment:
-      HUB_HOST: webdriver
+    web-base:
+        build: .
+        command: gunicorn --reload --bind=0.0.0.0:8000 eahub.config.wsgi
+        ports:
+            - 8000:8000
+        volumes:
+            - ./data:/data:rw
+        env_file:
+            - .env
+    web:
+        extends: web-base
+        depends_on:
+            db:
+                condition: service_healthy
+            mail:
+                condition: service_started
+        links:
+            - db:postgres
+        volumes:
+            - .:/app:rw
+    web-e2e:
+        extends: web-base
+        depends_on:
+            db:
+                condition: service_healthy
+            mail:
+                condition: service_started
+            chrome:
+                condition: service_started
+        links:
+            - db:postgres
+    db:
+        image: postgres:9.6
+        ports:
+            - 5432:5432
+        environment:
+            POSTGRES_DB: db
+            POSTGRES_HOST_AUTH_METHOD: trust
+        volumes:
+            - .:/app:rw
+        healthcheck:
+            test: ["CMD-SHELL", "pg_isready -U postgres"]
+            interval: 5s
+            timeout: 5s
+            retries: 7
+            start_period: 3s
```

Comment:
FYI, this is the fix for github actions postgres error - basically now github should wait for the container to start instead of just assuming that the db process is ready to accept connections